std/range/package.d

```diff
@@ -1626,6 +1653,78 @@ pure @safe unittest // issue 18657
     assert(equal(r, "foo"));
 }
 
+@safe unittest
+{
+    static void* p;
+    static struct R
+    {
+        void* q;
+        int front;
+        bool empty;
+        void popFront() {}
+        @property R save() { p = q; return this; }
+            // `p = q;` is only there to prevent inference of `scope return`.
+    }
+    R r;
+    choose(true, r, r).save;
+}
+
+// Make sure ChooseResult.save doesn't trust @system user code.
+@system unittest // copy is @system
```

Comment:
Shouldn't this have a `__traits(compiles` with @safe in there too ensure that this never gets safe?

Reply:
That's the next unittest.